server/src/modules/runs/runs.service.ts

```diff
@@ -1,7 +1,45 @@
-import { Injectable } from '@nestjs/common';
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { Prisma } from '@prisma/client';
+import { PaginatedResponseDto } from '../../@common/dto/paginated-response.dto';
+import { RunDto } from '../../@common/dto/run/runs.dto';
+import { ExpandToPrismaIncludes } from '../../@common/utils/dto.utility';
 import { RunsRepoService } from '../repo/runs-repo.service';
 
 @Injectable()
 export class RunsService {
     constructor(private readonly runRepo: RunsRepoService) {}
+
+    async getAll(
+        skip?: number,
+        take?: number,
+        mapID?: number,
+        userID?: number,
+        userIDs?: number[],
+        flags?: number,
+        expand?: string[]
+    ): Promise<PaginatedResponseDto<RunDto>> {
+        const where: Prisma.RunWhereInput = {};
+        const include: Prisma.RunInclude = ExpandToPrismaIncludes(
+            expand?.filter((x) => ['player', 'baseStats', 'zoneStats', 'rank', 'map'].includes(x))
+        );
+
+        if (expand?.includes('mapWithInfo')) include.map = { include: { info: true } };
+
+        if (userID && userIDs) throw new BadRequestException();
+
+        if (mapID) {
+            where.mapID = mapID;
+        }
+        if (userID) {
+            where.playerID = userID;
+        } else if (userIDs) {
+            where.playerID = { in: userIDs };
+        }
+        if (flags) {
+            where.flags = flags; // TODO: Change this to bitwise enum crap
```

Comment:
Let's leave a comment here mentioning that this will be expanded during 0.10.0. But not as a TODO, those should be for things within the rewrite only.